fs/hash/hash.go

```diff
@@ -248,6 +250,67 @@ func (m *MultiHasher) Size() int64 {
 	return m.size
 }
 
+// GetHashState returns the partial hash state for the given hash type encoded as a string
+func (m *MultiHasher) GetHashState(hashType Type) (string, error) {
+	h, ok := m.h[hashType]
+	if !ok {
+		return "", ErrUnsupported
+	}
+	marshaler, ok := h.(encoding.BinaryMarshaler)
```

Comment:
How many of our hashes implement `encoding.BinaryMarshaler`? I don't think any I've written do (dropboxhash, onedrive hash).

Reply:
stdlib hashes do like md5, sha1, sha256

Reply:
I will submit a pr for mrhash

Reply:
... and for dropbox/onedrive hashes after this pr gets merged

Reply:
Captured in https://github.com/rclone/rclone/issues/5624